Xamarin.Forms.Platform.UAP/Platform.cs

```diff
@@ -270,39 +270,49 @@ void OnRendererSizeChanged(object sender, SizeChangedEventArgs sizeChangedEventA
 
 		async void SetCurrent(Page newPage, bool popping = false, Action completedCallback = null)
 		{
-			if (newPage == _currentPage)
-				return;
+			try
+			{
+				if (newPage == _currentPage)
+					return;
 
 			if (_currentPage != null)
 			{
 				Page previousPage = _currentPage;
 				IVisualElementRenderer previousRenderer = GetRenderer(previousPage);
 				_container.Children.Remove(previousRenderer.ContainerElement);
 
-				if (popping)
-				{
-					previousPage.Cleanup();
-					// Un-parent the page; otherwise the Resources Changed Listeners won't be unhooked and the 
-					// page will leak 
-					previousPage.Parent = null;
+					if (popping)
+					{
+						previousPage.Cleanup();
+						// Un-parent the page; otherwise the Resources Changed Listeners won't be unhooked and the 
+						// page will leak 
+						previousPage.Parent = null;
+					}
 				}
-			}
 
-			newPage.Layout(ContainerBounds);
+				newPage.Layout(ContainerBounds);
 
-			IVisualElementRenderer pageRenderer = newPage.GetOrCreateRenderer();
-			_container.Children.Add(pageRenderer.ContainerElement);
+				IVisualElementRenderer pageRenderer = newPage.GetOrCreateRenderer();
+				_container.Children.Add(pageRenderer.ContainerElement);
 
-			pageRenderer.ContainerElement.Width = _container.ActualWidth;
-			pageRenderer.ContainerElement.Height = _container.ActualHeight;
+				pageRenderer.ContainerElement.Width = _container.ActualWidth;
+				pageRenderer.ContainerElement.Height = _container.ActualHeight;
 
-			completedCallback?.Invoke();
+				completedCallback?.Invoke();
 
-			_currentPage = newPage;
+				_currentPage = newPage;
 
-			UpdateToolbarTracker();
+				UpdateToolbarTracker();
 
-			await UpdateToolbarItems();
+				await UpdateToolbarItems();
+			}
+			catch(Exception error)
+			{
+				if (error.HResult == -2147417842)
```

Comment:
Where did the `-2147417842` constant come from? And what exactly does it mean?
Could you please add a comment with a description and a link to the docs. And also convert it to HEX => `0x8001010E`

Reply:
I did not do the conversion to not cost processing. Since the variable already sets int, I thought it best to just compare.